src/net.cpp

```diff
@@ -1080,7 +1080,12 @@ void ThreadSocketHandler()
                 }
             }
         }
-        if(vNodes.size() != nPrevNodeCount) {
+        size_t vNodesSize;
+        {
+            LOCK(cs_vNodes);
+            vNodesSize = vNodes.size();
+        }
+        if(vNodesSize != nPrevNodeCount) {
             nPrevNodeCount = vNodes.size();
```

Comment:
Line 1084 (old): `nPrevNodeCount = vNodes.size();`
to
Line 1089 (new): `nPrevNodeCount = vNodesSize;`

Otherwise you are still accessing `vNodes` without `LOCK`.